ember-core/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -542,11 +384,22 @@ private[ember] object Parser {
           idx += 1
         }
 
-        if (throwable != null) RespPreludeError("Encounterd Error parsing", Option(throwable))
-        if (httpVersion != null && status != null)
-          RespPreludeComplete(httpVersion, status, bv.drop(idx))
-        else RespPreludeIncomplete
+        if (throwable != null)
+          F.raiseError(RespPreludeError("Encountered Error parsing", Option(throwable)))
+        else if (httpVersion == null || status == null)
+          F.raiseError(RespPreludeError("Failed to parse HTTP response prelude", None))
+        else
+          RespPrelude(httpVersion, status, idx).pure[F]
       }
+
+      private val space = ' '.toByte
+      private val cr: Byte = '\r'.toByte
+      private val lf: Byte = '\n'.toByte
```

Comment:
I think this is maybe the third time I've seen these defined, but I guess it's not like they're big or going to change.

Reply:
Maybe they can be pulled into `Shared`